src/Context/SessionContext.php

```diff
@@ -127,10 +150,30 @@ public static function restore(array $sessionContextBody): SessionContext
             $sessionContextBody[self::FIELD_EXPIRY_TIME]
         );
         $sessionContext->userId = $sessionContextBody[self::FIELD_USER_ID];
+        $sessionContext->userPerson = static::getUserOrNull($sessionContextBody, self::FIELD_USER_PERSON);
+        $sessionContext->userCompany = static::getUserOrNull($sessionContextBody, self::FIELD_USER_COMPANY);
+        $sessionContext->userApiKey = static::getUserOrNull($sessionContextBody, self::FIELD_USER_API_KEY);
+        $sessionContext->userPaymentServiceProvider =
+            static::getUserOrNull($sessionContextBody, self::FIELD_USER_PAYMENT_SERVICE_PROVIDER);
 
         return $sessionContext;
     }
 
+    /**
+     * @param array $sessionContextBody
+     * @param string $userType
+     *
+     * @return UserPerson|UserCompany|UserApiKey|UserPaymentServiceProvider|null
+     */
+    private static function getUserOrNull(array $sessionContextBody, string $userType)
+    {
+        if (isset($sessionContextBody[$userType])) {
+            return UserPerson::createFromJsonString(json_encode($sessionContextBody[$userType]));
```

Comment:
@angelomelonas is this correct? This will always return a UserPerson, right?

Reply:
@NickvandeGroes you are right, I extracted some code and then missed this. Will fix!